app/models/search/bookmark_query.rb

```diff
@@ -224,7 +247,12 @@ def collections_filter
 
   def tag_exclusion_filter
     if exclusion_ids.present?
-      exclusion_ids.flatten.map { |exclusion_id| term_filter(:filter_ids, exclusion_id) }
+      exclusion_ids.flatten.map { |exclusion_id|
```

Comment:
There's a lot of flattens here. Maybe just use a single flat_map and rely on the fact that exclusion_ids will always return a list of IDs?